### PR TITLE
Made some enhancements on the Scala Options' code

### PR DESCRIPTION
**Pull request by mahmoudmahdi24**
The main goal of this Pull Request is to make some code enhancements on the different Options used in this project.
Some pattern matching snippets were unnecessary since we could have treated options as collections.
for example : 

- **foreach** can replace the following code: 
```
option match {
  case None => {}
  case Some(x) => foo(x)
}
```
- **forall** can replace the following code: 
```
option match {
  case None => true
  case Some(x) => foo(x)
}
```
- **exists** can replace the following code:
```
option match {
  case None => false
  case Some(x) => foo(x)
}
```

I also removed some unnecessary pattern matching conditions which could have been anticipated using a ```getOrElse``` on the Scala Option.

These modifications were tested by running the existant tests.

### CODE REVIEW

File: src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala [hunk 1]
```diff
@@ -36,8 +36,8 @@ case class DescribeDeltaHistoryCommand(
     val basePath =
       new Path(if (table.nonEmpty) {
         DeltaTableIdentifier(sparkSession, table.get) match {
-          case Some(id) if id.path.isDefined => id.path.get
-          case Some(id) => throw DeltaErrors.tableNotSupportedException("DESCRIBE HISTORY")
+          case Some(id) => id.path.getOrElse(
+            throw DeltaErrors.tableNotSupportedException("DESCRIBE HISTORY"))
```
Comment (tdas) [inline src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala]: we generally put `throw X` in `{ }` so make this, and at the other place `getOrElse { throw ... }`

**Comment by tdas** [comment]
this is good!

**Comment by mahmoudmahdi24** [comment]
Hello @tdas ,
Thanks a lot for your feedback. I made the requested changes.
Otherwise, it seems that there's some issues on Circle CI with the ```
Prepare Python env``` step (not only on my pipeline, but also on other pipelines). Can you please re-run my pipeline once the problem is fixed ? (I don't have the necessary privileges).
Thanks a lot,

File: src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala (hunk 1 above)
Comment (mahmoudmahdi24) [reply src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala]: done 👍

File: src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala [hunk 2]
```diff
@@ -36,8 +36,8 @@ case class DescribeDeltaHistoryCommand(
     val basePath =
       new Path(if (table.nonEmpty) {
         DeltaTableIdentifier(sparkSession, table.get) match {
-          case Some(id) if id.path.isDefined => id.path.get
-          case Some(id) => throw DeltaErrors.tableNotSupportedException("DESCRIBE HISTORY")
+          case Some(id) => id.path.getOrElse{
```
Comment (tdas) [inline src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala]: i think by the style guide there should be space after `getOrElse`
and `}` should in the line after

File: src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala [hunk 3]
```diff
@@ -44,8 +44,7 @@ case class VacuumTableCommand(
     val pathToVacuum =
       new Path(if (table.nonEmpty) {
         DeltaTableIdentifier(sparkSession, table.get) match {
-          case Some(id) if id.path.isDefined => id.path.get
-          case Some(id) => throw DeltaErrors.tableNotSupportedException("VACUUM")
+          case Some(id) => id.path.getOrElse{throw DeltaErrors.tableNotSupportedException("VACUUM")}
```
Comment (tdas) [inline src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala]: same here. there should be spaces. before and after `{` and `}`.

File: src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala (hunk 2 above)
Comment (mahmoudmahdi24) [reply src/main/scala/io/delta/tables/execution/DescribeDeltaHistoryCommand.scala]: done 👍

File: src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala (hunk 3 above)
Comment (mahmoudmahdi24) [reply src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala]: done 👍

File: src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala [hunk 4]
```diff
@@ -96,11 +96,9 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
   }
 
   private def subqueryNotSupportedCheck(condition: Option[Expression], op: String): Unit = {
-    condition match {
-      case Some(cond) if SubqueryExpression.hasSubquery(cond) =>
+    condition.foreach(cond => if (SubqueryExpression.hasSubquery(cond)) {
```
Comment (rxin) [inline src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala]: this should really be

```
condition.foreach { cond =>
  if (...) {
  }
}
```

File: src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala [hunk 5]
```diff
@@ -265,20 +265,16 @@ object SchemaUtils {
...
-              // if existing value is non-nullable, so should be the new value
-              && (existingField.nullable || !newField.nullable)
-              // and the type of the field must be compatible, too
-              && isDatatypeReadCompatible(existingField.dataType, newField.dataType))
-          case None =>
-            // new fields are fine, they just won't be returned
-            true
-        }
+        // new fields are fine, they just won't be returned
+        existing.get(newField.name).forall(existingField => {
```
Comment (rxin) [inline src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala]: here too

File: src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala (hunk 4 above)
Comment (mahmoudmahdi24) [reply src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala]: Done 👍

File: src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala (hunk 5 above)
Comment (mahmoudmahdi24) [reply src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala]: Done 👍

**Comment by mahmoudmahdi24** [comment]
Hello @rxin , Thanks for your feedback. I made the requested changes 👍

**Comment by tdas** [comment]
looks good to me, I will merge it soon.

**Pull request closed without merging**